screens/CreateGame.js

```diff
@@ -27,30 +30,61 @@ function CreateGame({ navigation }) {
   const [chosenDate, setChosenDate] = useState(new Date());
   const [showError, setShowError] = useState(false);
   const [errorMessage, setErrorMessage] = useState("");
+  const [locations, setLocations] = useState([]);
+  const [selectedPlayers, setSelected] = useState([]);
+
+  //get all locations to populate the location picker
+  async function getLocations() {
+    setLoading(true);
+    try {
+      const allLocations = await DataStore.query(Location);
+      //map the name to an array
+      const formatLocations = allLocations.map(({ name: value }) => ({
+        value,
+        label: value,
+      }));
+      setLocations(formatLocations);
+    } catch (error) {
+      //if locations don't load for some reason then we should probably display an error message and route back
+      console.log(error.message);
+    }
+    setLoading(false);
+  }
+
+  async function getPlayers() {
+    setLoading(true);
+    let toInvite = [];
+    try {
+      const allPlayers = await DataStore.query(Player, (p) => p.email.ne(thisPlayer.email));
+      allPlayers.map((element) => {
+        toInvite.push({label: element.name, value: element.id});
+      });
+      console.log("to invite: ", toInvite);
+      setToInvite(toInvite);
+    } catch (error) {
+      console.log("Error occurred in getting all players: " + error.message);
+    }
+  }
+  useEffect(() => {
+    getPlayer();
```

Comment:
You will need to await these functions when calling them. Call setLoading(true) before getPlayer() and setLoading(false) after getLocations(). This way you wont need to continuously call setLoading(true) and setLoading(false) in your helper functions. You can just call setLoading(false) inside all catch blocks. Maybe give those async functions a return value boolean so inside of the useEffect, we can determine whether the previous async function call was successful.

if (!(await getPlayer())) return;
if (!(await getPlayers())) return;
if (!await getLocations())) return;

Something like this.